include/sycl/algorithm/buffer_algorithms.hpp

```diff
@@ -287,7 +287,7 @@ B buffer_map2reduce(ExecutionPolicy &snp,
     cl::sycl::accessor<B, 1, cl::sycl::access::mode::read_write,
                        cl::sycl::access::target::local>
       sum { cl::sycl::range<1>(d.nb_work_item), cgh };
-    cgh.parallel_for_work_group<class wg>(rng, [=](cl::sycl::group<1> grp) {
+    cgh.parallel_for_work_group<class wg>(rng.get_global(), [=](cl::sycl::group<1> grp) {
```

Comment:
Maybe there is a better fix here but parallel_for_work_group can't take nd_range. I am not sure if we want to specify a work_group_size as well?